src/one/nio/util/DateParser.java

```diff
@@ -0,0 +1,201 @@
+package one.nio.util;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+
+/**
+ * Parser for subset of rfc2616 (http) and subset of ISO 1806 date formats.
+ */
+public class DateParser {
+
+    public static long parse(String input) {
+        if (input == null) {
+            throw new NullPointerException("Cannot provide null input to date parser.");
+        }
+        // simple detection for rfc2616
+        if (input.endsWith("GMT")) {
+            return parseRfc2616(input);
+        }
+        return parseIso1806(input);
+    }
+
+    private static long parseIso1806(String input) {
```

Comment:
Yes, made a typo with 1806 instead of 8601.

My use case is to parse an output from a C++ program. The C++ program encodes microseconds as part of the ISO8601 style date string. Example date string put in json by the C++ program. "2022-08-18T12:01:27.967875+0200".